README.md

```diff
@@ -20,51 +20,65 @@ Try out KaTeX [on the demo page](https://khan.github.io/KaTeX/#demo)!
+<html>
+  <head>
+    <link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/katex@0.10.0-beta/dist/katex.min.css" integrity="sha384-9tPv11A+glH/on/wEu99NVwDPwkMQESOocs/ZGXPoIiLE8MU/qkqUcZ3zzL+6DuH" crossorigin="anonymous">
+
+    <!-- The loading of KaTeX is deferred to speed up page rendering -->
+    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-beta/dist/katex.min.js" integrity="sha384-U8Vrjwb8fuHMt6ewaCy8uqeUXv4oitYACKdB0VziCerzt011iQ/0TqlSlv8MReCm" crossorigin="anonymous"></script>
+
+    <!-- To automatically render math in text elements, include the auto-render extension: -->
+    <script defer src="https://cdn.jsdelivr.net/npm/katex@0.10.0-beta/dist/contrib/auto-render.min.js" integrity="sha384-aGfk5kvhIq5x1x5YdvCp4upKZYnA8ckafviDpmWEKp4afOZEqOli7gqSnh8I6enH" crossorigin="anonymous"
+        onload="renderMathInElement(document.body);"></script>
+  </head>
+  ...
+</html>
 ```
 
+You can also [download KaTeX](https://github.com/khan/katex/releases) and host it yourself.
+
+For details on how to configure auto-render extension, refer to [the documentation](https://khan.github.io/KaTeX/docs/autorender.html).
+
+### API
+
 #### In-browser rendering
 
 Call `katex.render` with a TeX expression and a DOM element to render into:
 
 ```js
-katex.render("c = \\pm\\sqrt{a^2 + b^2}", element);
+try {
+    katex.render("c = \\pm\\sqrt{a^2 + b^2}", element);
+} catch (e) {
+    if (e instanceof katex.ParseError) {
+        // KaTeX can't parse the expression
+    } else {
+        throw e;
+    }
+}
 ```
 
 If KaTeX can't parse the expression, it throws a `katex.ParseError` error.
 
-#### Rendering expressions in text elements
-
-To automatically render math in text elements, include the [auto-render script](https://khan.github.io/KaTeX/docs/autorender.html) `contrib/auto-render.min.js`, or via CDN:
-
-```html
-<script src="https://cdn.jsdelivr.net/npm/katex@0.10.0-beta/dist/contrib/auto-render.min.js" integrity="sha384-aGfk5kvhIq5x1x5YdvCp4upKZYnA8ckafviDpmWEKp4afOZEqOli7gqSnh8I6enH" crossorigin="anonymous"></script>
-````
-
-Then, call the `renderMathInElement` function with a DOM element containing expressions in a script tag before the closing body tag:
-
-```html
-<body>
-  ...
-  <script>
-    renderMathInElement(document.body);
-  </script>
-</body>
-```
-
-See [Auto-render Extension](https://khan.github.io/KaTeX/docs/autorender.html) for more details.
-
 #### Server side rendering or rendering to a string
 
 To generate HTML on the server or to generate an HTML string of the rendered math, you can use `katex.renderToString`:
 
 ```js
-var html = katex.renderToString("c = \\pm\\sqrt{a^2 + b^2}");
-// '<span class="katex">...</span>'
+try {
+    var html = katex.renderToString("c = \\pm\\sqrt{a^2 + b^2}");
+    // '<span class="katex">...</span>'
+} catch (e) {
+    if (e instanceof katex.ParseError) {
+        // KaTeX can't parse the expression
+    } else {
+        throw e;
+    }
+}
```

Comment:
I think we should keep these examples simple in the README, and refer to the API doc for more.  I'd use the `{throwOnError: false}` option in both the examples above.  This also quickly demos the ability to pass in a settings object.